### PR TITLE
Default value for preloadPagesCount property and throwing errors on incorrect values

### LINKED ISSUE #2: Negative numbers in preloadPagesCount
Hey again,
Do you know what is happening when `preloadPagesCount` is negative? I tried with `preloadPagesCount = -1` and `-3` and in both cases it behaves like I'll just put a `1` - is it desired behavior?

Maybe a negative numbers in `preloadPagesCount` property should throw an error since those are not logically correct?

### LINKED ISSUE #3: Not providing the preloadPagesCount throw an Exception
When you update the library from 0.1.2 to 0.1.3 and you'll not provide the new `preloadPagesCount` property, widget will throw an exception:

```logcat
I/flutter (28504): ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
I/flutter (28504): The following NoSuchMethodError was thrown building Scrollable(axisDirection: right, physics:
I/flutter (28504): PageScrollPhysics, dirty, dependencies: [_LocalizationsScope-[GlobalKey#d9d95], _InheritedTheme,
I/flutter (28504): ScrollConfiguration], state: ScrollableState#cda16(position: _PagePosition#c976f(offset: null,
I/flutter (28504): range: null..null, viewport: null, ScrollableState, PageScrollPhysics -> ClampingScrollPhysics,
I/flutter (28504): IdleScrollActivity#f9ea2, ScrollDirection.idle))):
I/flutter (28504): The method '<=' was called on null.
I/flutter (28504): Receiver: null
I/flutter (28504): Tried calling: <=(1)
I/flutter (28504): 
I/flutter (28504): When the exception was thrown, this was the stack:
I/flutter (28504): #0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:50:5)
I/flutter (28504): #1      _PreloadPageViewState.build.<anonymous closure> (package:preload_page_view/preload_page_view.dart:554:45)
I/flutter (28504): #2      ScrollableState.build (package:flutter/src/widgets/scrollable.dart:551:27)
I/flutter (28504): #3      StatefulElement.build (package:flutter/src/widgets/framework.dart:3825:27)
I/flutter (28504): #4      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3736:15)
I/flutter (28504): #5      Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (28504): #6      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3716:5)
I/flutter (28504): #7      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3864:11)
I/flutter (28504): #8      ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (28504): #9      Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (28504): #10     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (28504): #11     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3747:16)
I/flutter (28504): #12     Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (28504): #13     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3716:5)
I/flutter (28504): #14     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (28504): #15     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (28504): #16     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (28504): #17     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3747:16)
I/flutter (28504): #18     Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (28504): #19     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3716:5)
I/flutter (28504): #20     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3864:11)
I/flutter (28504): #21     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3711:5)
I/flutter (28504): #22     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2956:14)
I/flutter (28504): #23     Element.updateChild (package:flutter/src/widgets/framework.dart:2759:12)
I/flutter (28504): #24     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3747:16)
I/flutter (28504): #25     Element.rebuild (package:flutter/src/widgets/framework.dart:3559:5)
I/flutter (28504): #26     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2273:33)
I/flutter (28504): #27     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:700:20)
I/flutter (28504): #28     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:268:5)
I/flutter (28504): #29     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:988:15)
I/flutter (28504): #30     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:928:9)
I/flutter (28504): #31     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:840:5)
I/flutter (28504): #35     _invoke (dart:ui/hooks.dart:209:10)
I/flutter (28504): #36     _drawFrame (dart:ui/hooks.dart:168:3)
I/flutter (28504): (elided 3 frames from package dart:async)
I/flutter (28504): ════════════════════════════════════════════════════════════════════════════════════════════════════
```

I suggest assigning `preloadPagesCount` some default value. I think `1` will be ok. When we agree to some default value I'll happy to create a pull request with this change.

### PR DESCRIPTION
**Pull request by wrbl606**
This resolves #2 with throwing proper exceptions in constructor and resolves #3 with a default value of 1. 0s are now allowed.

Also, I've updated the example app but it refers to my fork of this repo so it should be changed after merging this pull request.

### CODE REVIEW

File: .idea/libraries/Dart_Packages.xml [hunk 1]
```diff
@@ -5,182 +5,182 @@
         <entry key="async">
           <value>
             <list>
-              <option value="$PROJECT_DIR$/../../NeilLab/flutter/.pub-cache/hosted/pub.dartlang.org/async-2.0.8/lib" />
+              <option value="C:/flutter/.pub-cache/hosted/pub.dartlang.org/async-2.0.8/lib" />
```
Comment (Neil-Gu) [inline .idea/libraries/Dart_Packages.xml]: It's my bad but could you help to fix `git ignore` to ignore all `.idea` changes?

File: example/lib/main.dart [hunk 2]
```diff
@@ -30,14 +31,15 @@ class _PreloadPageViewState extends State<PreloadPageViewDemo> {
...
-        body: Center(
+        body: Container(
             child: PreloadPageView.builder(
-          itemCount: 3,
-          itemBuilder: (BuildContext context, int position) {
-            new Text("item" + position.toString());
-          },
+          preloadPagesCount: 5,
+          itemBuilder: (BuildContext context, int position) =>
+              DemoPage(position),
```
Comment (Neil-Gu) [inline example/lib/main.dart]: Looks good. Have you run this example in you phone?

File: lib/preload_page_view.dart [hunk 3]
```diff
@@ -515,13 +536,26 @@ class _PreloadPageViewState extends State<PreloadPageView> {
     _lastReportedPage = widget.controller.initialPage;
   }
 
+  void _validatePreloadPagesCount(int preloadPagesCount) {
+    if (preloadPagesCount == null) {
+      throw 'preloadPagesCount cannot be null!';
```
Comment (Neil-Gu) [inline lib/preload_page_view.dart]: 👍

File: lib/preload_page_view.dart [hunk 4]
```diff
@@ -515,13 +536,26 @@ class _PreloadPageViewState extends State<PreloadPageView> {
...
+    }
+
+    if (preloadPagesCount < 0) {
+      throw 'preloadPagesCount cannot be less than 0. Actual value: $_preloadPagesCount';
```
Comment (Neil-Gu) [inline lib/preload_page_view.dart]: 👍

File: lib/preload_page_view.dart [hunk 5]
```diff
@@ -551,15 +589,18 @@ class _PreloadPageViewState extends State<PreloadPageView> {
         physics: physics,
         viewportBuilder: (BuildContext context, ViewportOffset position) {
           return Viewport(
-            cacheExtent: _preloadPagesCount <= 1
-                ? 1
+            cacheExtent: _preloadPagesCount < 0
```
Comment (Neil-Gu) [inline lib/preload_page_view.dart]: 👍

File: .idea/libraries/Dart_Packages.xml (hunk 1 above)
Comment (wrbl606) [reply .idea/libraries/Dart_Packages.xml]: Sure thing, I'll thought about removing it too

File: lib/preload_page_view.dart (hunk 5 above)
Comment (Neil-Gu) [inline lib/preload_page_view.dart]: There are a issue here, it should be:
```
_preloadPagesCount < 1
? 0
: (_preloadPagesCount ==1 
    ? 1
    : ((widget.scrollDirection == Axis.horizontal
        ? MediaQuery.of(context).size.width * _preloadPagesCount - 1
        : MediaQuery.of(context).size.height * _preloadPagesCount -1),))
```
BUT, due to it became soooo complicate, perhaps we could create some variable and use some if statement to do that.

**Comment by Neil-Gu** [comment]
Thanks for your contribution. Minor comments otherwise LGTM

File: example/lib/main.dart (hunk 2 above)
Comment (wrbl606) [reply example/lib/main.dart]: Yes, it did work on my phone. Tests would be nice though, we can include those from Flutter team and add some for preload checking.

File: lib/preload_page_view.dart (hunk 5 above)
Comment (wrbl606) [reply lib/preload_page_view.dart]: I think a function like `_calculateCacheExtent()` that takes all of these variables into account and returns proper cacheExtent value would make the code cleaner.

**Comment by wrbl606** [comment]
Hey, I've fixed issues you mentioned and tested the example app for correct and incorrect values - it behaves like it should, but still - tests are welcome.

File: .idea/libraries/Dart_Packages.xml (hunk 1 above)
Comment (Neil-Gu) [reply .idea/libraries/Dart_Packages.xml]: 👍

File: example/lib/main.dart (hunk 2 above)
Comment (Neil-Gu) [reply example/lib/main.dart]: 👍

**Pull request merged**